release/v1alpha1/types.go

```diff
@@ -0,0 +1,181 @@
+// ReleaseConfig.  The ReleaseConfig is a definition of how releases are calculated.  When a ReleasePayload is
+// generated, it will be generated in the same namespace as the imagstream that produced it. If/when an update
+// occurs, to one of these imagestreams, the release-controller will:
+//   1) Create a point-in-time mirror of the updated imagestream
+//   2) Create a new Release from the mirror
+//        - Any errors before this point will cause the release to marked `Failed`
+//   3) Launches a set of release analysis jobs
+//   4) Launches an aggregation job
+//   5) Launches a set of release verification jobs
+//        - These can either be `Blocking Jobs` which will prevent release acceptance or `Informing Jobs` which will
+//          not prevent release acceptance.
+//   6) Monitors for job completions
+//        - If all `Blocking Jobs` complete successfully, then the release is `Accepted`.  If any `Blocking Jobs` fail,
+//          the release will be marked `Rejected`
+//   7) Publishes all results to the respective webpage
+//
+// Example:
+// ART:
+//   1) Publishes an update to the `ocp/4.9-art-latest` imagestream
+//
+// Release-controller:
+//   1) Creates a mirror named: `ocp/4.9-art-latest-2021-09-27-105859`
+//   2) Creates an OpenShift Release: `ocp/release:4.9.0-0.nightly-2021-09-27-105859`
+//   3) Launches: 4.9.0-0.nightly-2021-09-27-105859-aggregated-<name>-analysis-<count>
+//   4) Launches: 4.9.0-0.nightly-2021-09-27-105859-aggregated-<name>-aggregator
+//   5) Launches: 4.9.0-0.nightly-2021-09-27-105859-<name>
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type ReleasePayload struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec the inputs used to create the ReleasePayload
+	Spec ReleasePayloadSpec `json:"spec,omitempty"`
+
+	// Status is the current status of the ReleasePayload
+	Status ReleasePayloadStatus `json:"status,omitempty"`
+}
+
+// ReleasePayloadSpec has the information to represent a PromotionTest
+type ReleasePayloadSpec struct {
+	PayloadCoordinates PayloadCoordinates `json:"payloadCoordinates,omitempty"`
+}
+
+type PayloadCoordinates struct {
+	Namespace          string `json:"namespace,omitempty"`
+	ImagestreamName    string `json:"imagestreamName,omitempty"`
+	ImagestreamTagName string `json:"imagestreamTagName,omitempty"`
+}
+
+// ReleasePayloadStatus the status of all the promotion test jobs
+type ReleasePayloadStatus struct {
+	// Conditions communicates the state of the ReleasePayload.
+	//
+	// Supported conditions include PayloadCreated, PayloadCreationFailed, Accepted, and Rejected.
+	//
+	// If PayloadCreated is false the ReleasePayload is waiting for a release image to be created and pushed to the
```

Comment:
Since the imagestreamtagname is required in order to specify the name of this ReleasePayload, can this ever be false?  It's ok to have the explicit condition, but what would false mean here.